### DISCUSSION TITLE
Multiple Views

### DISCUSSION THREAD

**Discussion by zmkhan90** [freekmurze/laravel-analytics#445] [Ideas]
Hi,

I want to ask if I can use multiple Views and Multiple JSON Credentials?

**Comment by MarekNguyen** [comment]
Does anyone have the solutions for this? i have the same situation!

**Comment by wildlifechorus** [comment]
Bumping this, need it too

**Comment by elbotha18** [comment]
Hey all

I am using this package to get the data for multiple properties in my Laravel 10 project. In order to do so, I store all the property ids in my database. Then I set the property_id before calling the Analytics::get();

Example::
$properties = GaProperty::all();
foreach ($properties as $property)
{
// set config to use property id
config([
            'analytics.property_id' => $property->property_id,
        ]);

// get $data for the property
$data = Analytics::get(Period($days), $metrics, $dimensions);
// use $data as needed
}

Hope this works for you guys!

**Comment by xeboo** [comment]
Unfortunately it doesn't work for me :(

**Comment by budikhafi** [comment]
it can be done with using setPropertyId function on its facade, 

ex: 
`Analytics::setPropertyId($yourPropertyId)->fetchTotalVisitorsAndPageViews(Period::days(2));`
